src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts

```diff
@@ -948,19 +948,21 @@ export class SettingsEditor2 extends EditorPane {
 		return ancestors.reverse();
 	}
 
-	private updateChangedSetting(key: string, value: any): Promise<void> {
+	private updateChangedSetting(key: string, value: any, manualReset: boolean): Promise<void> {
 		// ConfigurationService displays the error if this fails.
 		// Force a render afterwards because onDidConfigurationUpdate doesn't fire if the update doesn't result in an effective setting value change
 		const settingsTarget = this.settingsTargetsWidget.settingsTarget;
 		const resource = URI.isUri(settingsTarget) ? settingsTarget : undefined;
 		const configurationTarget = <ConfigurationTarget>(resource ? ConfigurationTarget.WORKSPACE_FOLDER : settingsTarget);
 		const overrides: IConfigurationOverrides = { resource };
 
-		const isManualReset = value === undefined;
+		const configurationTargetIsWorkspace = configurationTarget === ConfigurationTarget.WORKSPACE || configurationTarget === ConfigurationTarget.WORKSPACE_FOLDER;
 
-		// If the user is changing the value back to the default, do a 'reset' instead
+		const isManualReset = configurationTargetIsWorkspace ? manualReset : value === undefined;
+
+		// If the user is changing the value back to the default, and we're not targeting a workspace scope, do a 'reset' instead
 		const inspected = this.configurationService.inspect(key, overrides);
-		if (inspected.defaultValue === value) {
+		if (!configurationTargetIsWorkspace && inspected.defaultValue === value) {
```

Comment:
If this turns out to be confusing, then another way to do it would be to do a reset for workspace scopes unless the setting is configured in a higher scope (in which case setting it to the default would be overriding the higher scope)

Reply:
@roblourens is this PR ready to merge?

Reply:
Yep. I think I meant to give a ✅  with the last review, sorry